math/complex_numbers.cpp

```diff
@@ -0,0 +1,190 @@
+     * @return result current number plus other number
+     */
+    Complex operator+(const Complex &other) {
+        Complex result(this->re + other.re, this->im + other.im);
+        return result;
+    }
+
+    /**
+     * Operator overload to be able to subtract two complex numbers.
+     * @param other The other number being subtracted from the current number.
+     * @return result current number subtract other number
+     */
+    Complex operator-(const Complex &other) {
+        Complex result(this->re - other.re, this->im - other.im);
+        return result;
+    }
+
+    /**
+     * Operator overload to be able to multiple two complex numbers.
+     * @param other The other number to multiply the current number to.
+     * @return result current number times other number.
+     */
+    Complex operator*(const Complex &other) {
+        Complex result(this->re * other.re - this->im * other.im,
+                       this->re * other.im + this->im * other.re);
+        return result;
+    }
+
+    /**
+     * Operator overload of the BITWISE NOT which gives us the conjugate of our
+     * complex number. NOTE: This is overloading the BITWISE operator but its
+     * not a BITWISE operation in this definition.
+     * @return result The conjugate of our complex number.
+     */
+    Complex operator~() const {
+        Complex result(this->re, -(this->im));
+        return result;
+    }
+
+    /**
+     * Operator overload to be able to divide two complex numbers. This function
+     * would throw an exception if the other number is zero.
+     * @param other The other number we divide our number by.
+     * @return result Current number divided by other number.
+     */
+    Complex operator/(const Complex &other) {
+        Complex result = *this * ~other;
+        double denominator = other.abs() * other.abs();
+        if (denominator != 0) {
+            result = Complex(result.real() / denominator,
+                             result.imag() / denominator);
+            return result;
+        } else {
+            throw std::invalid_argument("Undefined Value");
+        }
+    }
+};
+
+/**
+ * Logical Equal overload for our Complex class.
+ * @param a Left hand side of our expression
+ * @param b Right hand side of our expression
+ * @return 'True' If real and imaginary parts of a and b are same
+ * @return 'False' Otherwise.
+ */
+bool operator==(const Complex &a, const Complex &b) {
+    double del_real = a.real() - b.real();
+    double del_imag = a.imag() - b.imag();
+    return ((del_real <= 1e-15 && del_real >= -1e-15) &&
+            (del_imag <= 1e-15 && del_imag >= -1e-15));
+}
+
+/**
+ * Overloaded insersion operator to accommodate the printing of our complex
+ * number in their standard form.
+ * @param os The console stream
+ * @param num The complex number.
+ */
+std::ostream &operator<<(std::ostream &os, const Complex &num) {
+    os << num.real();
```

Comment:
maybe put the complex number within parentheses?

Reply:
Sure, will add that now.